### DISCUSSION TITLE
mysql connect strings

### DISCUSSION THREAD

**Discussion by bakerken** [hasura/graphql-engine#9722] [Question]
I'm having issues connecting to a public mysql instance: this works:

mysql -u anonymous -h mysql-eg-publicsql.ebi.ac.uk -D fusarium_oxysporum_core_56_109_2 -P 4157

There is no password.

I've got a freshly-activated trial version v2.27.0, and I've tried the following connect strings:
jdbc:mysql://mysql-eg-publicsql.ebi.ac.uk:4157/fusarium_oxysporum_core_56_109_2
jdbc:mysql://anonymous@mysql-eg-publicsql.ebi.ac.uk:4157/fusarium_oxysporum_core_56_109_2
jdbc:mysql://anonymous:@mysql-eg-publicsql.ebi.ac.uk:4157/fusarium_oxysporum_core_56_109_2
jdbc:mysql://anonymous:anonymous@mysql-eg-publicsql.ebi.ac.uk:4157/fusarium_oxysporum_core_56_109_2
jdbc:mysql://address=(host=mysql-eg-publicsql.ebi.ac.uk)(port=4157)(user=anonymous)/fusarium_oxysporum_core_56_109_2
jdbc:mysql://address=(host=mysql-eg-publicsql.ebi.ac.uk,port=4157,user=anonymous)

All fail with access denied. How should I use a null password?

**Comment by codedmart** [comment]
Can you try:
`jdbc:mysql://mysql-eg-publicsql.ebi.ac.uk:4157/fusarium_oxysporum_core_56_109_2?user=anonymous`

**Reply by bakerken** [reply]
Thanks; that apparently connected, but I got an error:

Error while fetching data
Error fetching GDC tables
